compiler/rustc_resolve/src/lib.rs

```diff
@@ -435,6 +436,11 @@ impl ModuleKind {
             ModuleKind::Def(.., name) => Some(*name),
         }
     }
+
+    /// Check if the module is an enum.
+    pub fn is_enum(&self) -> bool {
+        matches!(self, ModuleKind::Def(DefKind::Enum, _, _))
+    }
```

Comment:
Could you avoid this trivial method that is used once?

Reply:
I would rather keep it as it makes the code easier to understand and it will make it easier in the future if `ModuleKind` or `DefKind` are changed.

Reply:
Also, it's used once *here*, but it's something that will very possibly be useful later.

Reply:
This is pretty much the same situation as https://github.com/rust-lang/rust/pull/79956#discussion_r541699385.
"Later" does not exist right now, if it ever materializes the function could be added.

I'll just remove it when I touch this code next time (which is soon, because visibilities in resolve have some other issues), so I'd prefer it to not be introduced in the first place.

Reply:
I will remove it if you feel strongly about it, but there are already seem to be similar methods on `ModuleData`:

```rust
    // `self` resolves to the first module ancestor that `is_normal`.
    fn is_normal(&self) -> bool {
        matches!(self.kind, ModuleKind::Def(DefKind::Mod, _, _))
    }

    fn is_trait(&self) -> bool {
        matches!(self.kind, ModuleKind::Def(DefKind::Trait, _, _))
    }
```

Reply:
So do you still feel I should remove it?

Reply:
Yes, I thing removing is better.